test/unit/FormulateInputButton.test.js

```diff
@@ -78,3 +78,8 @@ test('type "button" renders slot inside button', () => {
   })
   expect(wrapper.find('button > span').html()).toBe('<span>My custom slot</span>')
 })
+
+it('passes an explicitly given name prop through to the root element', () => {
```

Comment:
It seems the intention around the new `hasGivenName` property in the context object is to allow the name to _not_ be present if it wasn't given. Which makes sense, names should be opt-in for security reasons (credit cards for example). Can you also add reciprocal tests to ensure names are not assigned if no name is given?

Reply:
Well, the issue here was that names are generated from IDs always, so they do exist. I only want to pass one down in the event that a user has actually put a `name` prop on a formulate input.

Reply:
Right, but the name `attribute` is not output unless you define one. At least it shouldn't be, so we need to ensure that remains true.

Reply:
Gotcha, ok yeah I’ll add those tests